### ISSUE TITLE
Build Failure with Xcode 9

### ISSUE THREAD

**Issue by mgaudet** [eclipse/omr#2015]
This is new instance of a problem we'd addressed before in #1707, and puts more pressure on #1758 

```
14:07 $ ninja -j1 comptest
[1/66] Building CXX object jitbuilder/CMakeFiles/jitbuilder.dir/__/compiler/optimizer/Structure.cpp.o
FAILED: jitbuilder/CMakeFiles/jitbuilder.dir/__/compiler/optimizer/Structure.cpp.o 
/usr/local/opt/ccache/libexec/c++  -DBITVECTOR_BIT_NUMBERING_MSB -DJ9HAMMER -DJITTEST -DOSX -DPROD_WITH_ASSUMES -DSUPPORTS_THREAD_LOCAL -DTR_HOST_64BIT -DTR_HOST_X86 -DTR_TARGET_64BIT -DTR_TARGET_X86 -DUT_DIRECT_TRACE_REGISTRATION -D_FILE_OFFSET_BITS=64 -I../jitbuilder/x/amd64 -I../jitbuilder/x -I../jitbuilder -I../compiler/x/amd64 -I../compiler/x -I../compiler -I../ -I. -I.././include -I.././include_core -I.././third_party -Wno-write-strings -std=c++0x -g -fPIC   -pthread -m64 -MD -MT jitbuilder/CMakeFiles/jitbuilder.dir/__/compiler/optimizer/Structure.cpp.o -MF jitbuilder/CMakeFiles/jitbuilder.dir/__/compiler/optimizer/Structure.cpp.o.d -o jitbuilder/CMakeFiles/jitbuilder.dir/__/compiler/optimizer/Structure.cpp.o -c ../compiler/optimizer/Structure.cpp
In file included from ../compiler/optimizer/Structure.cpp:33:
In file included from ../compiler/il/AliasSetInterface.hpp:28:
In file included from ../compiler/compile/SymbolReferenceTable.hpp:25:
In file included from ../compiler/compile/OMRSymbolReferenceTable.hpp:37:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:820:5: error: static_assert failed "Allocator::value_type must be same type as value_type"
    static_assert((is_same<typename allocator_type::value_type, value_type>::value),
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../compiler/optimizer/Structure.cpp:240:19: note: in instantiation of template class 'std::__1::map<TR_RegionStructure *, TR_BitVector, std::__1::less<TR_RegionStructure *>, TR::typed_allocator<std::__1::pair<TR_RegionStructure *, TR_BitVector>, TR::Region &> >' requested here
   RegionContents _regionContents;
                  ^
1 error generated.
ninja: build stopped: subcommand failed.
```

**Comment by mstoodle** [comment]
It's not a super helpful error message from the compiler either :( .

**Comment by mgaudet** [comment]
The lesson we learned from #1707 is that the answer is almost certainly just adding const to the allocator definition. Though #1758 is a hope that we could figure out a way to catch this in older build compilers.

**Comment by mgaudet** [comment]
Closed by #2031
